lxd/instance/drivers/driver_qemu.go

```diff
@@ -921,16 +927,45 @@ func (d *qemu) Start(stateful bool) error {
 		"-readconfig", confFile,
 		"-pidfile", d.pidFilePath(),
 		"-D", d.LogFilePath(),
-		"-chroot", d.Path(),
+	}
+
+	// If stateful, restore now.
+	if stateful {
+		if !d.stateful {
+			err = fmt.Errorf("Instance has no existing state to restore")
+			op.Done(err)
+			return err
+		}
+
+		qemuCmd = append(qemuCmd, "-incoming", "defer")
+	} else {
+		// The chroot option can't be used when restoring a migration stream.
+		qemuCmd = append(qemuCmd, "-chroot", d.Path())
+
+		if d.stateful {
+			// Stateless start requested but state is present, delete it.
+			err := os.Remove(d.StatePath())
+			if err != nil && !os.IsNotExist(err) {
+				op.Done(err)
+				return err
+			}
+
+			d.stateful = false
+			err = d.state.Cluster.UpdateInstanceStatefulFlag(d.id, false)
+			if err != nil {
+				op.Done(err)
+				return errors.Wrap(err, "Error updating instance stateful flag")
+			}
+		}
 	}
 
 	// SMBIOS only on x86_64 and aarch64.
 	if shared.IntInSlice(d.architecture, []int{osarch.ARCH_64BIT_INTEL_X86, osarch.ARCH_64BIT_ARMV8_LITTLE_ENDIAN}) {
 		qemuCmd = append(qemuCmd, "-smbios", "type=2,manufacturer=Canonical Ltd.,product=LXD")
 	}
 
-	// Attempt to drop privileges.
-	if d.state.OS.UnprivUser != "" {
+	// Attempt to drop privileges (doesn't work when restoring state).
+	if !stateful && d.state.OS.UnprivUser != "" {
```

Comment:
@stgraber should we issue a warning in the logs when starting up in privileged mode?

Reply:
Similar to the chroot comment. I don't think so as there's nothing the user can do about it.